src/components/input-panel/spec-editor/renderer.tsx

```diff
@@ -183,14 +132,14 @@ class Editor extends React.Component<Props, {}> {
   public componentWillUnmount() {
     document.removeEventListener('keydown', this.handleKeydown);
   }
-  public updateSpec(spec) {
+  public async updateSpec(spec) {
     let parsedMode = this.props.mode;
 
     try {
-      const schema = JSON.parse(spec).$schema;
+      const schemaUrl = JSON.parse(spec).$schema;
```

Comment:
I wonder whether we can put the logic below somewhere else where we have already parsed the spec into an object.